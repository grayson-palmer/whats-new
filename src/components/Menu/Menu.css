nav {
  color: #FFF;
  grid-column: 1;
  grid-row: 1 / span 2;
  
}

nav > * {
  margin: .6em 0;
}

nav h1 {
  font-size: 2em;
  text-align: center;
}

.new {
  color: #feae01;
}

ul li {
  margin: 1em 0;
  padding: .25em;
  text-align: left;
  font-size: 1.5em;
}

ul li:hover {
  background-color: #feae01;
  border-radius: 0 2.5em 2.5em 0;
}

.nav-category {
  display: flex;
  align-items: center;
}

.nav-category img {
  margin: .5em;
  height: auto;
  width: 2.5em;
}