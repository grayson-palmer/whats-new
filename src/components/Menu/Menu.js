import React, { Component } from 'react';
import './Menu.css';
import localImg from '../../location.png';
import techImg from '../../lightbulb.png';
import entImg from '../../tickets.png';
import sciImg from '../../rocket.png';
import healthImg from '../../health.png';

const Menu = (props) => {
  return (
    <nav>
      <h1>What's <span className='new'>New?</span></h1>
      <ul className='nav-items'>
        <li className='nav-category' onClick={() => props.handleChange('local')}><img src={localImg} alt='local news'></img>Local News</li>
        <li className='nav-category' onClick={() => props.handleChange('technology')}><img src={techImg} alt='technology'></img>Technology</li>
        <li className='nav-category' onClick={() => props.handleChange('entertainment')}><img src={entImg} alt='entertainment'></img>Entertainment</li>
        <li className='nav-category' onClick={() => props.handleChange('science')}><img src={sciImg} alt='science'></img>Science</li>
        <li className='nav-category' onClick={() => props.handleChange('health')}><img src={healthImg} alt='health'></img>Health</li>
      </ul>
    </nav>
  )
}

export default Menu;