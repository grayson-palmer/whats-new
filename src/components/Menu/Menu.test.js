import React from 'react';
import { shallow } from 'enzyme';
import Menu from './Menu';

describe('Menu', () => {
  it('should present on the dom', () => {
    const wrapper = shallow(<Menu 
      handleChange={jest.fn()}
    />);

    expect(wrapper).toMatchSnapshot();
  })
})