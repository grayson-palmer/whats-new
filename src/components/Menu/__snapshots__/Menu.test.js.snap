// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu  1`] = `
<nav>
  <h1>
    What's 
    <span
      className="new"
    >
      New?
    </span>
  </h1>
  <ul
    className="nav-items"
  >
    <li
      className="nav-category"
      onClick={[Function]}
    >
      <img
        alt="local news"
        src="location.png"
      />
      Local News
    </li>
    <li
      className="nav-category"
      onClick={[Function]}
    >
      <img
        alt="technology"
        src="lightbulb.png"
      />
      Technology
    </li>
    <li
      className="nav-category"
      onClick={[Function]}
    >
      <img
        alt="entertainment"
        src="tickets.png"
      />
      Entertainment
    </li>
    <li
      className="nav-category"
      onClick={[Function]}
    >
      <img
        alt="science"
        src="rocket.png"
      />
      Science
    </li>
    <li
      className="nav-category"
      onClick={[Function]}
    >
      <img
        alt="health"
        src="health.png"
      />
      Health
    </li>
  </ul>
</nav>
`;
